allennlp/training/callbacks/callback.py

```diff
@@ -0,0 +1,29 @@
+from typing import TypeVar, Generic
+
+from allennlp.common.registrable import Registrable
+
+TrainerType = TypeVar('TrainerType')  # pylint: disable=invalid-name
+
+class Callback(Registrable, Generic[TrainerType]):
+    # Lower priority comes first
+    priority = 0
+
+    def __call__(self, event: str, trainer: TrainerType) -> None:
+        raise NotImplementedError
+
+    def get_training_state(self) -> dict:
```

Comment:
Suggestion: mirroring Pytorch and calling these `get_state_dict` and `set_state_dict` will mean people immediately know what they do

Reply:
I agree with this.